### PR TITLE
feat: drop files anywhere, any tab

### LINKED ISSUE #240: Accept package.json drops all over the page
1. Pick a file
2. Drag it over the page
3. Miss the target by a few pixels
4. The app is gone, the json file is shown in the browser

Pages that accept files on drop generally accept them on the whole page, or at least block drops that were dropped on the wrong target.

I'd move these lines to the main App myself, but I don't know how to connect them to the `FileUploadControl`  myself:

https://github.com/npmgraph/npmgraph/blob/867ea561c440de204048b5c14df73aff66ead13a/components/InfoPane/FileUploadControl.tsx#L112-L114

### PR DESCRIPTION
**Pull request by fregante**
- Closes #240 

I also reorganized the code out of FileUploadControl as I had mentioned in #258 

## Start page

Screenshot 8

## Demo

https://github.com/user-attachments/assets/2a327449-0e53-4adb-b923-c28344c39b7d

### CODE REVIEW

File: components/InfoPane/FilePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  // Reset field
+  ev.target.value = '';
+}
+
+export default function FilePicker({ label }: { label: string }) {
+  return (
+    <label className="link" style={{ display: 'inline' }}>
+      {label}
+      <input type="file" hidden onChange={onSelect} accept=".json" />
+    </label>
```
Comment (fregante) [inline components/InfoPane/FilePicker.tsx:20]: New file-picker-only component. It appears as the single "Choose file" _link_

Screenshot

File: components/InfoPane/InfoPane.tsx [hunk 2]
```diff
@@ -27,10 +27,17 @@ export default function InfoPane(props: HTMLProps<HTMLDivElement>) {
...
-      <FileUploadControl />
+      <p>It also accepts package.json:</p>
+      <ul>
+        <li>Drag and drop a file anywhere on the page</li>
+        <li>Paste package.json (copied as text)</li>
+        <li>Paste package.json (copied as file)</li>
+        <li>
+          <FilePicker label="Choose file" /> from your computer
+        </li>
+      </ul>
```
Comment (fregante) [inline components/InfoPane/InfoPane.tsx:40]: The drop area is gone and replaced by extended instructions. I'm open to make further style improvements. Ideally these would be part of a dedicated "Documentation" section but I don't want to move the simple text-only examples too far away from the field.

File: components/useExternalInput.module.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
+.root::after,
```
Comment (fregante) [inline components/useExternalInput.module.scss:1]: The style is copied from GitHub's own overlay when dragging files over the "Code" tab in a repo

<table>
<tr>
	<th>GitHub
	<th>npmgraph
<tr>
	<td>Screenshot 7
	<td>Screenshot 6
</table>

File: components/InfoPane/InfoPane.tsx [hunk 4]
```diff
@@ -27,10 +27,17 @@ export default function InfoPane(props: HTMLProps<HTMLDivElement>) {
...
         </li>
-        <li>package.json pastes (copied as text or file)</li>
       </ul>
 
-      <FileUploadControl />
+      <p>It also accepts package.json:</p>
+      <ul>
+        <li>Drag and drop a file anywhere on the page</li>
+        <li>Paste package.json (copied as text)</li>
+        <li>Paste package.json (copied as file)</li>
```
Comment (broofa) [inline components/InfoPane/InfoPane.tsx:36]:
```suggestion
        <li>Paste package.json (file or text)</li>
```

File: components/InfoPane/InfoPane.tsx (hunk 2 above)
Comment (broofa) [reply components/InfoPane/InfoPane.tsx:40]: Works for me!

**Review by broofa — APPROVED** [review]
Great stuff, man!  Thank you so much!

One small nit, but otherwise it looks good.

**Pull request merged**